bve-build/src/main.rs

```diff
@@ -45,58 +50,112 @@ use structopt::StructOpt;
+        if err.to_string().contains("Finished") && s == "bve-native" {
+            println!("Dealing with cbindgen bug; clearing cache and regenerating");
+            clean();
+            generate_c_bindings(options);
+            return;
+        }
     }
+    panic!("cbindgen error: {}", err)
+}
 
-    if !parsed.no_bindgen {
-        let config = cbindgen::Config::from_file("bve-native/cbindgen.toml").unwrap();
+fn generate_c_bindings(options: &Options) {
+    let config = cbindgen::Config::from_file("bve-native/cbindgen.toml").unwrap();
 
-        {
-            // C
-            let mut config = config.clone();
-            config.language = Language::C;
-            *config.header.as_mut().expect("bve-native/cbindgen.toml needs a header") +=
-                "/* C API for BVE-Reborn high performance libraries. */";
-            cbindgen::Builder::new()
-                .with_crate("bve-native")
-                .with_config(config)
-                .generate()
-                .unwrap()
-                .write_to_file("bve-native/include/bve.h");
+    {
+        // C
+        let mut config = config.clone();
+        config.language = Language::C;
+        *config.header.as_mut().expect("bve-native/cbindgen.toml needs a header") +=
+            "/* C API for BVE-Reborn high performance libraries. */";
+        let result = cbindgen::Builder::new()
+            .with_crate("bve-native")
+            .with_config(config)
+            .generate();
+        match result {
+            Ok(bindings) => {
+                bindings.write_to_file("bve-native/include/bve.h");
+            }
+            Err(err) => handle_cbindgen_error(&err, options),
         }
-        {
-            // C++
-            let mut config = config;
-            config.language = Language::Cxx;
-            config.export.prefix = None;
-            *config.header.as_mut().expect("bve-native/cbindgen.toml needs a header") +=
-                "/* C++ API for BVE-Reborn high performance libraries. */";
-            config.trailer = Some(read_to_string("bve-native/include/bve_cpp.hpp").unwrap());
-            cbindgen::Builder::new()
-                .with_crate("bve-native")
-                .with_config(config)
-                .generate()
-                .unwrap()
-                .write_to_file("bve-native/include/bve.hpp");
+    }
+    {
+        // C++
+        let mut config = config;
+        config.language = Language::Cxx;
+        config.export.prefix = None;
+        *config.header.as_mut().expect("bve-native/cbindgen.toml needs a header") +=
+            "/* C++ API for BVE-Reborn high performance libraries. */";
+        config.trailer = Some(read_to_string("bve-native/include/bve_cpp.hpp").unwrap());
+        let result = cbindgen::Builder::new()
+            .with_crate("bve-native")
+            .with_config(config)
+            .generate();
+        match result {
+            Ok(bindings) => {
```

Comment:
Weird that your `Ok` branch has {} but your `Err` one doesn't, but I guess code formatting is hard

Reply:
Yeah I guess, I don't really notice things like that and try to not use parens if at all possible.